### DISCUSSION TITLE
Rendering clusters does not work

### DISCUSSION THREAD

**Discussion by znat** [JustFly1984/react-google-maps-api#3100] [Q&A]
Hi there,
I am trying to replicate the example here (https://react-google-maps-api-docs.netlify.app/#markerclusterer)
However clusters don't render.
If I remove the `MarkerClusterer` component and just render the marker it works just fine.
Any idea on what I can improve in this code (copied below for easier reference)?

Using:
- React 18.x
- react-google-maps-api 2.12.2

```react
const { GoogleMap, LoadScript, MarkerClusterer, Marker } = require('../../')
const ScriptLoaded = require('../../docs/ScriptLoaded').default

const mapContainerStyle = {
  height: '400px',
  width: '800px',
}

const center = { lat: -28.024, lng: 140.887 }

const locations = [
  { lat: -31.56391, lng: 147.154312 },
  { lat: -33.718234, lng: 150.363181 },
  { lat: -33.727111, lng: 150.371124 },
  { lat: -33.848588, lng: 151.209834 },
  { lat: -33.851702, lng: 151.216968 },
  { lat: -34.671264, lng: 150.863657 },
  { lat: -35.304724, lng: 148.662905 },
  { lat: -36.817685, lng: 175.699196 },
  { lat: -36.828611, lng: 175.790222 },
  { lat: -37.75, lng: 145.116667 },
  { lat: -37.759859, lng: 145.128708 },
  { lat: -37.765015, lng: 145.133858 },
  { lat: -37.770104, lng: 145.143299 },
  { lat: -37.7737, lng: 145.145187 },
  { lat: -37.774785, lng: 145.137978 },
  { lat: -37.819616, lng: 144.968119 },
  { lat: -38.330766, lng: 144.695692 },
  { lat: -39.927193, lng: 175.053218 },
  { lat: -41.330162, lng: 174.865694 },
  { lat: -42.734358, lng: 147.439506 },
  { lat: -42.734358, lng: 147.501315 },
  { lat: -42.735258, lng: 147.438 },
  { lat: -43.999792, lng: 170.463352 },
]

const options = {
  imagePath:
    'https://developers.google.com/maps/documentation/javascript/examples/markerclusterer/m', // so you must have m1.png, m2.png, m3.png, m4.png, m5.png and m6.png in that folder
}

function createKey(location) {
  return location.lat + location.lng
}

const MapWithMarkerClusterer = () => {
  return (
    <ScriptLoaded>
      <GoogleMap id='marker-example' mapContainerStyle={mapContainerStyle} zoom={3} center={center}>
        <MarkerClusterer options={options}>
          {(clusterer) =>
            locations.map((location) => (
              <Marker key={createKey(location)} position={location} clusterer={clusterer} />
            ))
          }
        </MarkerClusterer>
      </GoogleMap>
    </ScriptLoaded>
  )
}
```

**Comment by kacper005** [comment]
Hi,

Same here. I had an type error first. Then i have just used <Fragment></Fragment> to get rid of this. It worked for the type error. But i still can't see my Markers. It seems like the live demo in docks is broke too https://react-google-maps-api-docs.netlify.app/#markerclusterer. I have been searching for a solution but i can't find anything. 


My code:

`<Map
					mapContainerStyle={{ width: '100%', height: '100%' }}
					center={center}
					zoom={3}
				>
					<MarkerClusterer>
						{(clusterer) => (
							<Fragment>
								{filterUnits(units).map((unit) => (
									<MarkerF
										key={unit.id}
										position={{
											lat: unit.latitude,
											lng: unit.longitude
										}}
										clusterer={clusterer}
									/>
								))}
							</Fragment>
						)}
					</MarkerClusterer>
				</Map>`

Btw if u use React v.18 i suggest to use <MarkerF/> instead. 
import { MarkerF } from '@react-google-maps/api';
